src/promptflow/tests/sdk_cli_test/recording_utilities/record_storage.py

```diff
@@ -159,7 +158,88 @@ def _recursive_create_hashable_args(self, item):
         else:
             return item
 
-    def set_record(self, input_dict: Dict, output) -> None:
+    def _parse_output_generator(self, output):
```

Comment:
add some docstring as it has some important and complicate logic here

Reply:
Of course